### DISCUSSION TITLE
Couldn't Manage to work Exclude Locations

### DISCUSSION THREAD

**Discussion by mertbakir** [valhalla/valhalla#3293] [Q&A] [answered]
I was trying "exclude_locations" option on routing service API. Yet, Valhalla goes exactly on the path that I excluded. How should I proceed? What could be wrong? There are no errors in the logs. 

```
valhalla_json = {"sources": sources, "targets": targets, "costing": "auto", "units": "km",
                     "exclude_locations": [{"lat": 38.626383, "lon": 34.718745},
                                           {"lat": 38.626533, "long": 34.719586},
                                           {"lat": 38.625771, "lon": 34.717199},
                                           {"lat": 38.666429, "lon": 34.743906},
                                           {"lat" :38.6671, "lon": 34.74256}]}
```

excluded points are inside the red rectangles.

![image](https://user-images.githubusercontent.com/43188411/131011335-3924a218-1f8d-40d1-8789-8dd851bfbc90.png)

**Comment by nilsnolde** [comment, marked as answer]
Are you using a more than 2-day old „latest“ tag of our (gis-ops) Valhalla image? See here: https://github.com/gis-ops/docker-valhalla/issues/29. Updating your local image should do the job.

Just 2 days I tried I the last time and it worked as it should. I tried routing, not matrix, but shouldn’t matter, they share the same functions for exclude_polygons.

**Reply by mertbakir** [reply]
Yes, it was the older image. ~~Now, I'm getting an error with the new image, container doesn't start:~~

```
2021/08/27 09:11:15.792705 [INFO] Tile extract successfully loaded with tile count: 2013
2021/08/27 09:11:15.792959 [WARN] (stat):  No such file or directory
2021/08/27 09:11:15.792992 [WARN] Traffic tile extract could not be loaded
terminate called after throwing an instance of 'boost::wrapexcept<boost::property_tree::ptree_bad_path>'
  what():  No such node (service_limits.max_avoid_locations.max_matrix_distance)
/valhalla/scripts/run.sh: line 54:    32 Aborted                 (core dumped) valhalla_service ${CUSTOM_CONFIG} ${server_threads}
```

**Reply by mertbakir** [reply]
@nilsnolde I got it working now. Thanks.

Just to clarify, if there is no other possible path when a polygon/location is excluded, it returns 400 and no path, right? Is this behaviour configurable, e.g. if there is no other option, you can use the excluded path? Should I check the status and send another request?

**Comment by nilsnolde** [comment]
Yep, that’s the clients job to retry in this case.
